### DISCUSSION TITLE
How to access Storybook globals from components inside .mdx file?

### DISCUSSION THREAD

**Discussion by kerryj89** [storybookjs/storybook#26589] [Help]
### Summary

I've created an `.mdx` file and inside it I created some React wrapper components that wrap the Storybook doc blocks to extend its functionality - works great!

But I want these wrapper components to be able to set/get from Storybook globals defined in `preview.tsx`'s `globalTypes`.

I am trying to leverage Storybook's globals so that it can re-render optimally, but should I instead be using localStorage and useEffect to watch for changes there for these wrapper components?  Sorry if my terminology or thinking is wrong, still learning React on the side.

**Comment by andreszapatavff** [comment]
@kerryj89 did you get anywhere with this?

**Reply by kerryj89** [reply]
It's been a while but I don't think so.  A quick search for `globalTypes` in my codebase with Storybook didn't return anything.  I did feel like I was getting somewhere using their [channels](https://storybook.js.org/docs/addons/configure-addons#channels)

The code in this [comment](https://github.com/storybookjs/testing-react/issues/30#issuecomment-1654647933) might lead you somewhere but maybe outside of the React system.

**Comment by andreszapatavff** [comment]
@kerryj89 thank you for getting back. I will check the link out.

**Comment by Sidnioulz** [comment]
https://github.com/storybookjs/storybook/discussions/32362#discussioncomment-14277252 from a duplicate conversation provides a working solution. Note that globals aren't directly reactive but according to the OP on that thread, MDX gets remounted entirely on globals changes anyway.
